### ISSUE TITLE
'az vm extension image list --latest' fails

### ISSUE THREAD

**Issue by johanburati** [Azure/azure-cli#9460]
### Version
`azure-cli                         2.0.65`  from the Azure Cloud Shell

### Bug
```
$ az vm extension image list --latest -l westcentralus -o tsv
The command failed with an unexpected error. Here is the traceback:

list index out of range
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 560, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 618, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 611, in _run_job
    six.reraise(*sys.exc_info())
  File "/opt/az/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 588, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 297, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 461, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/vm/custom.py", line 1272, in list_vm_extension_images
    cmd.cli_ctx, publisher_name, name, version, image_location, latest)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/vm/_actions.py", line 155, in load_extension_images_thru_services
    t.result()  # don't use the result but expose exceptions from the threads
  File "/opt/az/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/opt/az/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/opt/az/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/vm/_actions.py", line 137, in _load_extension_images_from_publisher
    'version': versions[0].name})
IndexError: list index out of range
```

### Details
Using the `az vm extension image list` command with the `--latest` flag generate a traceback, without the flag it returns the expected output.

**Comment by aktzbn** [comment]
Have the same stack with `az pipelines build definition list` and _azure-cli_ of version _2.0.66_.

**Comment by adewaleo** [comment]
> Have the same stack with `az pipelines build definition list` and _azure-cli_ of version _2.0.66_.

Looks like this is a problem with the [azure devops extension.](https://docs.microsoft.com/en-us/cli/azure/ext/azure-devops/?view=azure-cli-latest) Please can you open an issue [here (devops extension repo)](https://github.com/Azure/azure-devops-cli-extension) or [here (general cli extensions repo)](https://github.com/Azure/azure-cli-extensions)?

**Comment by atbagga** [comment]
@adewaleo It is a similar stack with same error but not the same issue. It is entirely different code base.

From the stack it looks like.. 
The error is in this function [_load_extension_images_from_publisher ](https://github.com/Azure/azure-cli/blob/9ad476d1911b2c46351f142c6112516d66e2af85/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py#L116) where for some case versions is an empty array and you are trying to access the first index. versions[0].

@aktzbn The issue with az pipelines build list was when user provides a organization name istead of url in the --org param in any devops extension command. This is now fixed in version 0.11.0 release ETA this week.

**Comment by adewaleo** [comment]
> @adewaleo It is a similar stack with same error but not the same issue. It is entirely different code base.
> 
> From the stack it looks like..
> The error is in this function [_load_extension_images_from_publisher ](https://github.com/Azure/azure-cli/blob/9ad476d1911b2c46351f142c6112516d66e2af85/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_actions.py#L116) where for some case versions is an empty array and you are trying to access the first index. versions[0].
> 
> @aktzbn The issue with az pipelines build list was when user provides a organization name istead of url in the --org param in any devops extension command. This is now fixed in version 0.11.0 release ETA this week.

@atbagga i was responding to the pipelines issue. 

I agree that the original comment / issue was regarding _load_extension_images_from_publisher which I have a pr out for. 

For some reason there is at least one extension image type that doesn't have associated versions.
